src/hardware/sblaster.cpp

```diff
@@ -605,14 +619,29 @@ static void PlayDMATransfer(Bitu size)
 
 static void SuppressInitialDMATransfer(Bitu size)
 {
-	// Only suppress the first dword-transfer or less
-	if (size <= sizeof(uint32_t) || !sb.speaker) {
+	// Always suppress if no DMA mode is set (cannot be interpretted)
+	if (sb.dma.mode == DSP_DMA_NONE) {
```

Comment:
Right; it conflicts with the desire to avoid suppressing either of the 16-bit DMA modes, because in if we're using a 16-bit DMA mode, then by definition we're using the SB16 card type, which defaults to the ppeaker being disabled or off.